deployment/ccip/changeset/cs_prerequisites.go

```diff
@@ -620,9 +651,189 @@ func deployPrerequisiteContracts(e cldf.Environment, ab cldf.AddressBook, state
 				e.Logger.Errorw("Failed to deploy factory burn mint erc20", "chain", chain.String(), "err", err)
 				return err
 			}
+
+			factoryBurnMintERC20 = factoryBurnMintERC20Contract.Contract // set this here so that the address can be referenced later
 		} else {
 			e.Logger.Infow("factory burn mint erc20 already deployed", "chain", chain.String(), "addr", factoryBurnMintERC20.Address)
 		}
+		if burnMintTokenPool == nil {
```

Comment:
minor nit: would it make it better to move the token pool deployment contracts to an utility method? for readability, this method is too big now, you can do it in a follow up too